stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/ChatDomainImpl.kt

```diff
@@ -170,6 +170,8 @@ internal class ChatDomainImpl internal constructor(
     internal val job = SupervisorJob()
     internal var scope = CoroutineScope(job + DispatcherProvider.IO)
 
+    internal var isRepositoryInitialized = false
```

Comment:
Do we really need this? Seems like `user.value != null` means the same

Reply:
At this point, yes, it is the same. But we would be connecting something that don't need to be connected. At a later moment, we could decide that the initialization of RepositoryFacede is not inside `setUser`, by using `user.value != null` we could create a hidden bug in the future.

Reply:
1. In my opinion, the boolean flag`isRepositoryInitialized` is a source of future bugs. It's always easy to forget to set it, to rewrite it wrongly, and so on. 
If you want to check that the repo has been initialized with data for some particular user you can do it. When we create stub-version in `createNoOpRepos` we don' pass to `RepositoryFacadeBuilder` any value for current user and we do it when we create real repository. You can use it, pass new field to `RepositoryFacade::currenUser: User?` and check if it's null or no for the same reason
2. IMO thinking about situations in the future when we want to distinguish the process of `setUser` and initializing of the repository is too early optimization. And it's a known common evil practice in CS. Let's do it when we really want. Right now the logic of our `ChatDomain` doesn't divide situations when the user is set and his data is cached in repo. IMO exposing `initRepositoryFacade` is not right, it's violation of encapsulation for ChatDomain

Reply:
Ok, removing the flag